gitbook-docs/market/economics.md

```diff
@@ -16,109 +18,164 @@ Proof Market constants:
+* _Withdrawn_: the order was canceled before matching, a wrong proof was submitted, or
+no proofs were submitted in the requested time interval.
+
+### Request/proposal matching process
+
+The Proof Market provides a more complex behavior than one-hop trading.
+Each deal includes the proof generation stage.
+This stage may result in a completed deal if proof was provided and verified by the Proof Market
+or a restart of the matching if no proof has been provided or proof's verification has failed.
+Moreover, some applications require generating proofs in the specified time interval.
+In the future, the following parameters will influence the matching algorithm:
+
+* Proof generation time.
+Proof producers provide time bounds _t\_a_ in which they can generate proof for the statement.
+Users can set the desired proof generation time _t\_b_.
+Only proposals with the _t\_a_ ≤ _t\_b_ can be matched with the user’s request.
+* Proof producers rating.
+Proof producer's rating depends on the rate of completed orders — orders where the proof producer
+has provided correct proof in the requested time interval.
+
+Here are the basic rules of order matching for a request with specified time _t\_b_ and price _p\_b_:
+
+1. Choose proposals with _t\_a_ ≤ _t\_b_ and _p\_a_ ≤ _p\_b_.
+2. Sort them by price and proof generation time.
+The user’s request defines the priority of these parameters.
 3. Apply proof producer's rating to the sorted list.
-4. Choose the top ask from the list.
+4. Choose the top proposal from the list.
 
-### Bids & Asks resale
+### Requests & proposals resale
 
-When creating a new bid, there may be situations when there is already an equivalent bid in the database. Two bids are considered equivalent if they match in the **statement** and **public\_input** fields. Therefore, when adding a new bid, the application first tries to find an equivalent bid in the system. If such a bid is found and there is already proof for it, the following happens:
+There may be situations when an equivalent request is already in the database.
+Two requests are considered equivalent if they match in the **statement** and **public\_input** fields.
+Upon receiving a new request, the application tries to find an equivalent request in the system.
+This is what happens if such a request is found and there's already proof for it:
 
-* the new bid immediately goes into the **completed** state;
-* the proof requester pays a commission to the proof market;
-* the proof requester pays a commission to the proof producer who generated the proof for the existing bid;
-* the proof requester pays a commission to the owner of the existing bid;
-* unspent tokens are returned to the proof requester.
+* The new request immediately goes into the **completed** state.
+* The proof requester pays commissions to the Proof Market, the proof producer who generated the proof
+for the existing request, and the owner of that request.
```

Comment:
who is the owner of the request? Some new role?

Reply:
How about “the proof buyer/requester of the original request”?
Btw, in this article we have terms “proof buyer” and “proof requester” for the same entity, should I fix that?